store/src/java/com/zimbra/cs/index/ZimbraTopDocs.java

```diff
@@ -65,7 +65,7 @@ public static ZimbraTopDocs create(int totalHits, List<ZimbraScoreDoc> scoreDocs
      * Create equivalent ZimbraTopDocs object to a Lucene TopDocs object
      */
     public static ZimbraTopDocs create(TopDocs luceneTopDocs) {
-        return new ZimbraTopDocs(luceneTopDocs.totalHits,
+        return new ZimbraTopDocs((int) luceneTopDocs.totalHits,
```

Comment:
How difficult would it be use long instead?  Would seem better to fit with the underlying type.  Just a minor push back as I assume it would be kind of scary if we blew an int here.

Reply:
I actually originally tried to propagate the long, but it seems to be more trouble than it's worth if we want to avoid the down-cast anywhere. To quote https://issues.apache.org/jira/browse/LUCENE-7872:

```Even though a single index cannot have more than 2B documents, TopDocs.merge might merge TopDocs instances that have more than 2B documents in total.```

In other words, an `int` is enough to hold the number of results from a single shard. And since we are co-locating mailbox documents, it's not actually possible at this time to have an overflow here, since we'd never be merging results from different shards.